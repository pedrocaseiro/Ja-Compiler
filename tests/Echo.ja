class teste {
    ;
    public static int main (String[] asd) {}
    ;
    public static double teste(boolean z) {
        z = z + 1;
    }
    ;
    public static boolean ifsANDcenas(int n, double c) {
        flag = true;
        if (a > b) {
            ;
            print(b);
            print(a);
        } else {
            ;;;;;

            print();
            print(a,b,c);

            if (x > a){};

            System.out.println("asd");
            System.out.println(a);
        }

        if (a < b)
            ;

        if (a < b)
            ;
        else
            ;

        if (a < b)
            a = b;
        else
            ;

        if (a < v)
            a = b;
        else
            b = a;

        if (a < v) {
            a = b;
        } else
            ;

        if (a < v) {
            a = b;
        } else {
            b = a;
        }

        if (a < b) {
            a = b;
            b = c;
        } else
            ;

        if (a < b) {
            a = b;
            b = c;
        } else
            b = a;

        if (a < b) {
            a = b;
            b = c;
        } else {
            b = a;
            b = c;
        }

        a = -(-(-5));
    }
    ;
    public static void whiles(String[] yo) {
        while (a < b)
            ;
        while (a < b) {
            ;
        }
        while (a < b) {}

        while(a < b) {
            a = b;
        }

        while(a < b) {
            a = b;
            b = a;
        }

        do
            ;
        while(a < b);
        do {
            ;
        } while(a < b);
        do {}
        while(a < b);

        do {
            a = b;
        } while(a < b);

        do {
            a = b;
            b = a;
        } while(a < b);
    }
    ;
    public static void ifprec() {
        if (a < b)
            if (b < c)
                b = c;
        else
            a = c;
    }
    ;
    public static void last(int FINAL) {
        {
            a = b;
            b = c;
        }
        int z,x;
        {
            print(a);
        }
    }
    ;
    ?
}
