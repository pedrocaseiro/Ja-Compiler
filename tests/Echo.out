Line 133, col 5: illegal character (?)
Program
..Id(teste)
..MethodDecl
....MethodHeader
......Int
......Id(main)
......MethodParams
........ParamDecl
..........StringArray
..........Id(asd)
....MethodBody
..MethodDecl
....MethodHeader
......Double
......Id(teste)
......MethodParams
........ParamDecl
..........Bool
..........Id(z)
....MethodBody
......Assign
........Id(z)
........Add
..........Id(z)
..........DecLit(1)
..MethodDecl
....MethodHeader
......Bool
......Id(ifsANDcenas)
......MethodParams
........ParamDecl
..........Int
..........Id(n)
........ParamDecl
..........Double
..........Id(c)
....MethodBody
......Assign
........Id(flag)
........BoolLit(true)
......If
........Gt
..........Id(a)
..........Id(b)
........Block
..........Call
............Id(print)
............Id(b)
..........Call
............Id(print)
............Id(a)
........Block
..........Call
............Id(print)
..........Call
............Id(print)
............Id(a)
............Id(b)
............Id(c)
..........If
............Gt
..............Id(x)
..............Id(a)
............Block
............Block
..........Print
............StrLit("asd")
..........Print
............Id(a)
......If
........Lt
..........Id(a)
..........Id(b)
........Block
........Block
......If
........Lt
..........Id(a)
..........Id(b)
........Block
........Block
......If
........Lt
..........Id(a)
..........Id(b)
........Assign
..........Id(a)
..........Id(b)
........Block
......If
........Lt
..........Id(a)
..........Id(v)
........Assign
..........Id(a)
..........Id(b)
........Assign
..........Id(b)
..........Id(a)
......If
........Lt
..........Id(a)
..........Id(v)
........Assign
..........Id(a)
..........Id(b)
........Block
......If
........Lt
..........Id(a)
..........Id(v)
........Assign
..........Id(a)
..........Id(b)
........Assign
..........Id(b)
..........Id(a)
......If
........Lt
..........Id(a)
..........Id(b)
........Block
..........Assign
............Id(a)
............Id(b)
..........Assign
............Id(b)
............Id(c)
........Block
......If
........Lt
..........Id(a)
..........Id(b)
........Block
..........Assign
............Id(a)
............Id(b)
..........Assign
............Id(b)
............Id(c)
........Assign
..........Id(b)
..........Id(a)
......If
........Lt
..........Id(a)
..........Id(b)
........Block
..........Assign
............Id(a)
............Id(b)
..........Assign
............Id(b)
............Id(c)
........Block
..........Assign
............Id(b)
............Id(a)
..........Assign
............Id(b)
............Id(c)
......Assign
........Id(a)
........Minus
..........Minus
............Minus
..............DecLit(5)
..MethodDecl
....MethodHeader
......Void
......Id(whiles)
......MethodParams
........ParamDecl
..........StringArray
..........Id(yo)
....MethodBody
......While
........Lt
..........Id(a)
..........Id(b)
........Block
......While
........Lt
..........Id(a)
..........Id(b)
........Block
......While
........Lt
..........Id(a)
..........Id(b)
........Block
......While
........Lt
..........Id(a)
..........Id(b)
........Assign
..........Id(a)
..........Id(b)
......While
........Lt
..........Id(a)
..........Id(b)
........Block
..........Assign
............Id(a)
............Id(b)
..........Assign
............Id(b)
............Id(a)
......DoWhile
........Block
........Lt
..........Id(a)
..........Id(b)
......DoWhile
........Block
........Lt
..........Id(a)
..........Id(b)
......DoWhile
........Block
........Lt
..........Id(a)
..........Id(b)
......DoWhile
........Assign
..........Id(a)
..........Id(b)
........Lt
..........Id(a)
..........Id(b)
......DoWhile
........Block
..........Assign
............Id(a)
............Id(b)
..........Assign
............Id(b)
............Id(a)
........Lt
..........Id(a)
..........Id(b)
..MethodDecl
....MethodHeader
......Void
......Id(ifprec)
......MethodParams
....MethodBody
......If
........Lt
..........Id(a)
..........Id(b)
........If
..........Lt
............Id(b)
............Id(c)
..........Assign
............Id(b)
............Id(c)
..........Assign
............Id(a)
............Id(c)
........Block
..MethodDecl
....MethodHeader
......Void
......Id(last)
......MethodParams
........ParamDecl
..........Int
..........Id(FINAL)
....MethodBody
......Block
........Assign
..........Id(a)
..........Id(b)
........Assign
..........Id(b)
..........Id(c)
......VarDecl
........Int
........Id(z)
......VarDecl
........Int
........Id(x)
......Call
........Id(print)
........Id(a)
..MethodDecl
....MethodHeader
......Int
......Id(empty)
......MethodParams
....MethodBody
